src/Oscilloscope.js

```diff
@@ -0,0 +1,49 @@
+class Oscilloscope {
+    constructor(context) {
+      console.log('Oscilloscope constructed')
+      this.context = context
+      this.analyzer = context.createAnalyser()
+    }
+
+    start(canvas) {
+      this.analyzer.fftSize = 2048
+      const analyzerBufferLength = this.analyzer.frequencyBinCount
+      const analyzerDataArray = new Uint8Array(analyzerBufferLength)
+
+      const canvasWidth = canvas.width
+      const canvasHeight = canvas.height
+      const canvasContext = canvas.getContext('2d')
+      const boundAnalyzer = this.analyzer
+
+      const draw = function() {
```

Comment:
If you use arrow syntax (`=>`) you shouldn't need to bind the analyzer on like 16. and you can access `this.analyzer` directly.

Also you should define the `draw` function outside of the `start` function as its own method of the class.

Reply:
I see what you mean about arrow syntax, but these two things are apparently mutually exclusive. You can't use arrow functions as class methods (and that is not their intended use anyways). See this SO: https://stackoverflow.com/questions/31095710/methods-in-es6-objects-using-arrow-functions

To use a hard bound function for draw, it must be declared inside of another class method. I'm pushing a commit that will move it to the constructor instead of the start method, and bind it with fat arrow so that the analyzer doesn't have to be bound manually. This does make the whole thing look cleaner I think.

This ES6 stuff is new to me, so if I'm misunderstanding something here, or this still doesn't look right to you, let me know and I'll take another look.